File: frontend/src/update_urls.py
import os
import re

files_to_update = [
  ('features/auth/AuthContext.tsx', 2),
  ('features/channels/ChannelSidebar.tsx', 2),
  ('features/chat/ChatArea.tsx', 2),
  ('features/servers/InviteLandingPage.tsx', 2),
  ('features/servers/InviteModal.tsx', 2),
  ('features/servers/MembersSidebar.tsx', 2),
  ('features/servers/ServerSidebar.tsx', 2),
  ('features/socket/SocketContext.tsx', 2),
  ('layouts/AppLayout.tsx', 1),
  ('pages/Login.tsx', 1),
  ('pages/Register.tsx', 1)
]

for file_path, depth in files_to_update:
    if not os.path.exists(file_path):
        print(f"Skipping {file_path}")
        continue
        
    with open(file_path, 'r') as f:
        content = f.read()
        
    if "API_BASE_URL" in content:
        continue

    content = content.replace('http://localhost:5000', '${API_BASE_URL}')
    content = content.replace("'${API_BASE_URL}", "`${API_BASE_URL}")
    content = content.replace("${API_BASE_URL}'", "${API_BASE_URL}`")
    
    import_path = '../' * depth + 'config'
    import_stmt = f"import {{ API_BASE_URL }} from '{import_path}';\n"
    
    # find last import
    imports = re.finditer(r'^import .*?;?\n', content, re.MULTILINE)
    last_import_idx = 0
    for match in imports:
        last_import_idx = match.end()
        
    content = content[:last_import_idx] + import_stmt + content[last_import_idx:]
    
    with open(file_path, 'w') as f:
        f.write(content)

print("Update complete")

File: frontend/src/update_urls.js
const fs = require("fs");
const path = require("path");

const filesToUpdate = [
  { path: "features/auth/AuthContext.tsx", depth: 2 },
  { path: "features/channels/ChannelSidebar.tsx", depth: 2 },
  { path: "features/chat/ChatArea.tsx", depth: 2 },
  { path: "features/servers/InviteLandingPage.tsx", depth: 2 },
  { path: "features/servers/InviteModal.tsx", depth: 2 },
  { path: "features/servers/MembersSidebar.tsx", depth: 2 },
  { path: "features/servers/ServerSidebar.tsx", depth: 2 },
  { path: "features/socket/SocketContext.tsx", depth: 2 },
  { path: "layouts/AppLayout.tsx", depth: 1 },
  { path: "pages/Login.tsx", depth: 1 },
  { path: "pages/Register.tsx", depth: 1 },
];

filesToUpdate.forEach((file) => {
  const fullPath = path.join(__dirname, file.path);
  let content = fs.readFileSync(fullPath, "utf8");

  // Replace http://localhost:5000 with ${API_BASE_URL}
  content = content.replace(/http:\/\/localhost:5000/g, "${API_BASE_URL}");
  // Fix quotes if it was in single quotes instead of backticks
  content = content.replace(/'\$\{API_BASE_URL\}/g, "`${API_BASE_URL}");
  content = content.replace(/\$\{API_BASE_URL\}'/g, "${API_BASE_URL}`");

  // Determine relative import path
  const importPath = "../".repeat(file.depth) + "config";

  // Add import statement if not already there
  if (!content.includes("API_BASE_URL")) {
    // it means replacement failed, or we already did it
  } else if (!content.includes("import { API_BASE_URL }")) {
    const importStmt = `import { API_BASE_URL } from '${importPath}';\n`;
    // Find the last import statement
    const importRegex = /import\s+.*?;?\n/g;
    let match;
    let lastIndex = 0;
    while ((match = importRegex.exec(content)) !== null) {
      lastIndex = match.index + match[0].length;
    }
    content =
      content.slice(0, lastIndex) + importStmt + content.slice(lastIndex);
  }

  fs.writeFileSync(fullPath, content);
});
